frontend/server/src/main/java/org/pytorch/serve/util/GPUManager.java

```diff
@@ -0,0 +1,159 @@
+package org.pytorch.serve.util;
+
+import java.util.Map;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.nio.charset.StandardCharsets;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class GPUManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(GPUManager.class);
+
+    private static GPUManager instance;
+
+    private int nGPUs;
+    private int minFreeMemory;
+    private float maxShareFailures;
+    private AtomicInteger[] nFailures;
+    private AtomicInteger[] freeMemory;
+    private ConcurrentHashMap<String, Integer> workerIds;
+
+    private GPUManager(int nGPUs, int minFreeMemory, float maxShareFailures) {
+        this.nGPUs = nGPUs;
+        this.minFreeMemory = minFreeMemory;
+        this.maxShareFailures = maxShareFailures;
+
+        if (nGPUs > 0) {
+            this.workerIds = new ConcurrentHashMap<String, Integer> ();
```

Comment:
```suggestion
            this.workerIds = new ConcurrentHashMap<> ();
```
If you want you can omit these parameters and Java will figure them out automatically.